monero-rpc/src/rpc/wallet.rs

```diff
@@ -313,6 +335,28 @@ impl Client {
         let r: Response<Refreshed> = serde_json::from_str(&response)?;
         Ok(r.result)
     }
+
+    /// Transfers the complete balance of the account to `address`.
+    pub async fn sweep_all(&self, address: &str) -> Result<SweepAll> {
+        let params = SweepAllParams {
+            address: address.into(),
+        };
+        let request = Request::new("sweep_all", params);
+
+        let response = self
+            .inner
+            .post(self.url.clone())
+            .json(&request)
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        debug!("sweep_all RPC response: {}", response);
+
+        let r: Response<SweepAll> = serde_json::from_str(&response)?;
```

Comment:
I usually try to avoid type hints and use turbofish syntax instead because it is more refactoring friendly, i.e.:

```rust
serde_json::from_str::<Response<SweepAll>(&response)
```

This way, your are co-locating the type with the expression that requires it. This makes it easier to take this expression and compose it with other ones without having to touch the type hint on the variable.

For example, it can easily transform this code:

```rust
let r = serde_json::from_str::<Response<SweepAll>(&response);

Ok(r.result)
```

into this:


```rust
let result = serde_json::from_str::<Response<SweepAll>(&response).result;

Ok(result)
```

and the type inference of the rust compiler will do the rest.

Specifying type hints gets in the way of this, which is why I only place them as a last resort.

Reply:
👍 I will add it to the follow-up cleaning that removes the traits (because it should be changed throughout the file)

Reply:
https://github.com/comit-network/xmr-btc-swap/pull/266